gnovm/pkg/gnolang/preprocess.go

```diff
@@ -157,24 +157,19 @@ func initStaticBlocks(store Store, ctx BlockNode, bn BlockNode) {
 			switch n := n.(type) {
 			case *AssignStmt:
 				if n.Op == DEFINE {
-					var defined bool
 					for _, lx := range n.Lhs {
 						nx := lx.(*NameExpr)
 						ln := nx.Name
 						if ln == blankIdentifier {
 							continue
 						}
 						if !isLocallyDefined2(last, ln) {
```

Comment:
maybe this one should only consider if it's defined in the block, rather than any other parent block. I'm wondering whether it could cause issues for a "shadowed" variable somehow not being predefined as normal?

But I don't know the full implications, just something to think about.